File: Day-01/ruby/Taucher2003/day_01.rb
input = ARGF.read
return if input == nil

elf_data = input.split("\n\n")
elf_calories = elf_data.map { |data| data.split("\n").map(&:to_i).sum }

puts "Part 1: #{elf_calories.max}"

puts "Part 2: #{elf_calories.sort.reverse.take(3).sum}"

File: Day-01/elixir/vKxni/day_01.exs
defmodule DayOne do
  def part_one() do
    input()
    |> Enum.map(fn x -> Enum.map(x, &String.to_integer/1) end)
    |> Enum.map(&Enum.sum(&1))

    |> Enum.max()
    |> IO.puts()
  end

  def part_two() do
    input()
    |> Enum.map(fn x -> Enum.map(x, &String.to_integer/1) end)
    |> Enum.map(&Enum.sum(&1))

    |> Enum.sort()
    |> Enum.reverse()
    |> Enum.take(3)
    |> Enum.sum()
    |> IO.puts()
  end

  defp input() do
    File.read!("input.txt")
    |> String.split("\n\n")
    |> Enum.map(&String.trim/1)
    |> Enum.map(&String.split(&1, "\n"))
  end
end

DayOne.part_one()
DayOne.part_two()
